go-app-ussd_higherhealth_healthcheck.js

```diff
@@ -1701,32 +1702,81 @@ go.app = (function () {
+      return new JsonApi(self.im).get(
+        // use dictionary params
+        self.im.config.eventstore.url + "/api/v3/covid19triage/", {
+        headers: {
+          "Authorization": ["Token " + self.im.config.eventstore.token],
+          "User-Agent": ["Jsbox/HH-Covid19-Triage-USSD"]
+        },
+        params: {
+          "timestamp_gt": date + "T00:00:00+02:00",
+          "msisdn": msisdn
+        }
+      }).then(function (response) {
+        var results = response.data.results;
+        if(results.length === 0){
+          if (self.im.user.answers.returning_user){
+            question = $([
+              "Welcome back to HIGHER HEALTH's HealthCheck.",
+              "No result SMS will be sent. Continue or WhatsApp HI to 0600110000",
+              "",
+              "Reply"
+            ].join("\n"));
+          }
+          else {
+            question = $([
+              "HIGHER HEALTH HealthCheck is your risk assessment tool.",
+              "No result SMS will be sent. Continue or WhatsApp HI to 0600110000",
+              "",
+              "Reply"
+            ].join("\n"));
+          }
+          return new MenuState(name, {
+            question: question,
+            error: $("This service works best when you select numbers from the list"),
+            accept_labels: true,
+            choices: [
+              new Choice("state_terms", $("START"))
+            ]
+          });
+        }
+        var last_result = results[results.length - 1];
+        self.im.user.answers.risk = last_result.risk;
+        self.im.user.answers.state_first_name = last_result.first_name;
+        self.im.user.answers.state_last_name = last_result.last_name;
         question = $([
           "Welcome back to HIGHER HEALTH's HealthCheck.",
           "No result SMS will be sent. Continue or WhatsApp HI to 0600110000",
           "",
           "Reply"
         ].join("\n"));
-      } else {
-        question = $([
-          "HIGHER HEALTH HealthCheck is your risk assessment tool.",
-          "No result SMS will be sent. Continue or WhatsApp HI to 0600110000",
-          "",
-          "Reply"
-        ].join("\n"));
-      }
-      return new MenuState(name, {
-        question: question,
-        error: $("This service works best when you select numbers from the list"),
-        accept_labels: true,
-        choices: [
-          new Choice("state_terms", $("START"))
-        ]
+        return new MenuState(name, {
+          question: question,
+          error: $("This service works best when you select numbers from the list"),
+          accept_labels: true,
+          choices: [
+            new Choice("state_terms", $("START")),
+            new Choice("state_display_risk", $("RECEIPT"))
+          ]
+        });
+      }, function (e) {
+        if(_.get(e, "response.code") === 404) {
+          self.im.user.answers = {returning_user: false};
+          return self.states.create("state_welcome");
+        }
```

Comment:
I don't think that this request will ever return a 404, it will just return a 200 with an empty list